OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -358,7 +358,7 @@ void RefreshServerList()
 
 				lanGames = lanGames.GroupBy(gs => gs.Address).Select(g => g.Last()).ToList();
 
-				Game.RunAfterTick(() => RefreshServerListInner(games.Concat(lanGames).ToList()));
+				Game.RunAfterTick(() => RefreshServerListInner(games == null ? (lanGames.Count == 0 ? null : lanGames.ToList()) : games.Concat(lanGames).ToList()));
```

Comment:
Ok, in that case you can still get most of the readability and efficiency improvements with something like:
```csharp
var groupedLanGames = lanGames.GroupBy(gs => gs.Address).Select(g => g.Last());
if (games != null)
    games = games.AddRange(groupedLanGames);
else
    games = groupedLanGames.ToList();

Game.RunAfterTick(() => RefreshServerListInner(games));
```